data_steward/gcloud/gcs/__init__.py

```diff
@@ -16,7 +16,21 @@ class StorageClient(Client):
     See https://googleapis.dev/python/storage/latest/client.html
     """
 
-    def list_sub_prefixes(self, bucket: str, prefix: str):
+    def empty_bucket(self, bucket: str, **kwargs) -> None:
+        """
+        Delete all blobs in a bucket.
+        :param name: A GCS bucket name.
+
+        Some common keyword arguments:
+        :param prefix: (Optional) Prefix used to filter blobs.
+        (i.e gsutil rm -r gs://bucket/prefix/)
+        """
+
+        blobs = self.list_blobs(bucket, **kwargs)
```

Comment:
One thing to consider here is how to handle buckets with potentially thousands, if not hundreds of thousands of records.  Attempting to retrieve the entire list of blogs in a single request could run us into issues if we try to operate on a bucket of sufficient size.

To that end, I'd recommend having a look at the `page` and `page_size` parameters, see how we could chunk up the total list by only having a few specific ones in memory at a time.

Reply:
OUTSTANDING point!  There is a max_results kwarg that maybe passed to address this 🙂
So for these use cases, you would probably provide a loop:
```
sc = StorageClient()
while sc.list_blobs('humongous_bucket', max_results=1):
    sc.empty_bucket('humongous_bucket', max_results=10000)
```

See: https://googleapis.dev/python/storage/latest/client.html#google.cloud.storage.client.Client.list_blobs for additional args

you might also address this using offsets to "page" through results:

start_offset (str) – (Optional) Filter results to objects whose names are lexicographically equal to or after startOffset. If endOffset is also set, the objects listed will have names between startOffset (inclusive) and endOffset (exclusive).

end_offset (str) – (Optional) Filter results to objects whose names are lexicographically before endOffset. If startOffset is also set, the objects listed will have names between startOffset (inclusive) and endOffset (exclusive).

Reply:
`max_results` _could_ do what we need if we assume that the page resets each time we delete a set of them.

I would also recommend having a look at the `batch()` functionality of the client: https://googleapis.dev/python/storage/latest/client.html#google.cloud.storage.client.Client.batch

This would help us execute fewer requests in total, as well.

Reply:
I'm not sure what needs to change.  Do you want the function to perform deletions as part of a batch operation/page approach?

Reply:
Ah, I see this:
https://github.com/googleapis/google-cloud-python/issues/727

and went with the pages you recommended above.  good catch.